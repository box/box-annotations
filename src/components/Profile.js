import PropTypes from 'prop-types';
import React from 'react';
import Avatar from 'box-react-ui/lib/components/avatar/Avatar';
import * as constants from '../constants';

/* eslint-disable require-jsdoc */
const ProfileContainer = ({ createdBy, name, id, avatarUrl }) => (
    <div>
        <div className={constants.CLASS_PROFILE_IMG_CONTAINER}>
            <Avatar id={id} name={name} avatarUrl={avatarUrl} />
        </div>
        <div className={constants.CLASS_PROFILE_CONTAINER}>
            <div className={constants.CLASS_USER_NAME}>{name}</div>
            <div className={constants.CLASS_COMMENT_DATE}>{createdBy}</div>
        </div>
    </div>
);

ProfileContainer.displayName = 'Profile';
ProfileContainer.propTypes = {
    /** classname to add to the container element. */
    avatarUrl: PropTypes.string,
    /** Url to avatar image.  If passed in, component will render the avatar image instead of the initials */
    createdBy: PropTypes.string,
    /** classname to add to the container element. */
    className: PropTypes.string,
    /** Users full name */
    name: PropTypes.string,
    /** Users full name */
    id: PropTypes.string
};

export default ProfileContainer;
