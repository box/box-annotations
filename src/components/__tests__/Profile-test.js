import * as React from 'react';
import { shallow } from 'enzyme';

import Profile from '../Profile';

describe('components/Profile', () => {
    test('should display user profile with created by date', () => {
        const user = {
            id: '123',
            name: 'Kanye West',
            avatarUrl: 'foo.bar'
        };
        const date = '2017-09-27T10:40:41-07:00';

        const wrapper = shallow(<Profile user={user} createdBy={date} />);
        expect(wrapper).toMatchSnapshot();
    });
});
