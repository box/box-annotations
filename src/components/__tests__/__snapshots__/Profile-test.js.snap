// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Profile should display user profile with created by date 1`] = `
<div>
  <Avatar
    avatarUrl="foo.bar"
    className="profile-image-container"
    id="123"
    name="Kanye West"
  />
  <div
    className="profile-container"
  >
    <div
      className="user-name"
    >
       
      Kanye West
       
    </div>
    <div
      className="comment-date"
    >
       
      2017-09-27T10:40:41-07:00
       
    </div>
  </div>
</div>
`;
